File: java/flyweight/exercise/WordProcessor.java
package flyweight.exercise;

public class WordProcessor {
    private TextElementFactory textElementFactory;

    public WordProcessor() {
        textElementFactory = new TextElementFactory();
    }

    public void renderText(String text, Font font, Color color, int x, int y) {
        // Iterate through characters in the text and render them using flyweight characters
        for (int i = 0; i < text.length(); i++) {
            char character = text.charAt(i);
            TextElement textElement = textElementFactory.getTextElement(character, font, color);
            textElement.render(x + i * font.getWidth(), y); // Adjust position based on font metrics
        }
    }
}

File: C#/Flyweight/Exercise/WordProcessor.cs
namespace Flyweight.Exercise;

public sealed class WordProcessor
{
    private readonly TextElementFactory _textElementFactory;

    public WordProcessor()
        => _textElementFactory = new();

    public void RenderTexts(string text, Font font, Color color, int x, int y)
    {
        // Iterate through characters in the text and render them using flyweight characters
        for (int i = 0; i < text.Length; i++)
        {
            char character = text[i];
            ITextElement textElement = _textElementFactory.GetTextElement(character, font, color);
            textElement.Render(x + i * font.Width, y);
        }
    }
}
